torchao/quantization/quant_api.py

```diff
@@ -279,10 +282,28 @@ def replace_conv2d_1x1(conv):
         model, replace_conv2d_1x1, filter_fn=filter_fn
     )
 
+def _quantization_type(weight: torch.Tensor):
+    if isinstance(weight, AffineQuantizedTensor):
+        return f"{weight.__class__.__name__}({weight._quantization_type()})"
+
+    if isinstance(weight, LinearActivationQuantizedTensor):
+        return f"{weight.__class__.__name__}(activation={weight.input_quant_func}, weight={_quantization_type(weight.original_weight_tensor)})"
+
+    if type(weight) is torch.Tensor:
+        return "not quantized"
+
+    return "not recognized"
 
-def _get_linear_subclass_inserter(constructor):
+def _linear_extra_repr(self):
+    return f"in_features={self.weight.shape[1]}, out_features={self.weight.shape[0]}, weight={_quantization_type(self.weight)}"
+
+def _get_linear_subclass_inserter(constructor, **kwargs):
+    """Helper function to apply the constructor that quantizes the weight Tensor (with additional kwargs)
+    to the weight of linear module
+    """
     def insert_subclass(lin):
-        lin.weight = torch.nn.Parameter(constructor(lin.weight), requires_grad=False)
+        lin.weight = torch.nn.Parameter(constructor(lin.weight, **kwargs), requires_grad=False)
+        lin.extra_repr = types.MethodType(_linear_extra_repr, lin)
```

Comment:
optional: is there anything we can do to preserve any preexisting custom repr?

Reply:
oh OK, I can add that